src/components/game/GameSettings.js

```diff
@@ -35,25 +60,82 @@ export default class GameSettings extends React.Component {
 
   setDifficultyHandler(difficulty) {
     this.setState({
-      difficulty: difficulty
+      difficulty: difficulty,
+      gameSetting: 'selectAmount'
+    })
+
+    if (this.state.category !== "any") {
+      const param = `total_${difficulty}_question_count`;
+      const availableQuestionsCount = this.state.availableQuestionsList[param];
+      this.setState({
+        availableQuestions: availableQuestionsCount
+      })
+    }
+  }
+
+  generateAPIRequestURL() {
+    const url = this.props.url
+      + "api.php?token=" + this.state.token
+      + (this.state.category === 'any' ? "" : "&category=" + this.state.category)
+      + (this.state.difficulty === 'any' ? "" : "&difficulty=" + this.state.difficulty)
+      + "&amount=" + (this.state.amount !== "" ? this.state.amount : "10");
+    //+ "&type=" + this.state.type; // to do: probably it's better to skip type, because num of questions are limited and there is no choise to check how many questions are available for multuple type game
```

Comment:
okay I thought true/false question was error when I found it on deploy-preview. make sense!